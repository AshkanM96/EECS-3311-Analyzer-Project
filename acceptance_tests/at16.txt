divide
integer_constant(-100)
integer_constant(-1)
type_check
evaluate
divide
reset
divide
type_check
evaluate
integer_constant(-100)
integer_constant(89)
evaluate
reset
subtract
subtract
integer_constant(-10)
integer_constant(1)
integer_constant(-10)
type_check
evaluate
