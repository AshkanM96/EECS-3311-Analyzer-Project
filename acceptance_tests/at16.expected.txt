  Expression currently specified: ?
  Report: Expression is initialized.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(-100)
  Expression currently specified: (-100 / ?)
  Report: OK.
->integer_constant(-1)
  Expression currently specified: (-100 / -1)
  Report: OK.
->type_check
  Expression currently specified: (-100 / -1)
  Report: (-100 / -1) is type-correct.
->evaluate
  Expression currently specified: (-100 / -1)
  Report: 100
->divide
  Expression currently specified: (-100 / -1)
  Report: Error (Expression is already fully specified).
->reset
  Expression currently specified: ?
  Report: OK.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->type_check
  Expression currently specified: (? / nil)
  Report: Error (Expression is not yet fully specified).
->evaluate
  Expression currently specified: (? / nil)
  Report: Error (Expression is not yet fully specified).
->integer_constant(-100)
  Expression currently specified: (-100 / ?)
  Report: OK.
->integer_constant(89)
  Expression currently specified: (-100 / 89)
  Report: OK.
->evaluate
  Expression currently specified: (-100 / 89)
  Report: -1
->reset
  Expression currently specified: ?
  Report: OK.
->subtract
  Expression currently specified: (? - nil)
  Report: OK.
->subtract
  Expression currently specified: ((? - nil) - nil)
  Report: OK.
->integer_constant(-10)
  Expression currently specified: ((-10 - ?) - nil)
  Report: OK.
->integer_constant(1)
  Expression currently specified: ((-10 - 1) - ?)
  Report: OK.
->integer_constant(-10)
  Expression currently specified: ((-10 - 1) - -10)
  Report: OK.
->type_check
  Expression currently specified: ((-10 - 1) - -10)
  Report: ((-10 - 1) - -10) is type-correct.
->evaluate
  Expression currently specified: ((-10 - 1) - -10)
  Report: -1
